File: 作品集/game/Parallel shiftシューティング/New Unity Project 12/Assets/syu-/aa/Script/Scripts/Bullet.cs
using UnityEngine;
using System.Collections;

public class Bullet : MonoBehaviour {

	//衝突判定
	void OnCollisionEnter(Collision collision){
		//衝突したものが箱だったら
		if(collision.gameObject.tag == "Box"){
			//ダメージメッセージを送信する
			collision.gameObject.SendMessage ("ApplyDamage");
		}
		//弾丸自体の消滅
		Destroy(gameObject);
	}
}

File: 作品集/game/弾幕ギャラクシー/New Unity Project 9/Assets/script/Bullet.js
//弾処理
function OnTriggerEnter (collisionInfo : Collider) {
    if (collisionInfo.gameObject.tag == "Player") {
        // プレイヤーにダメージを与える
        collisionInfo.gameObject.SendMessage("ApplyDamage");
    } else if (collisionInfo.gameObject.tag == "Bullet") {
        Destroy(gameObject);
    } else {
        Destroy(gameObject);
    }
}